src/components/Personalization/dom-actions/executeActions.js

```diff
@@ -43,15 +45,16 @@ const executeAction = (logger, modules, type, args) => {
 
 export default (actions, modules, logger) => {
   const actionPromises = actions.map(action => {
-    const { type } = action;
+    const processedAction = preprocessAction(action);
```

Comment:
Optionally, You can really rename this helper to `preprocess` and the `action` param would make the whole thing very readable :)

Reply:
Renamed to `preprocess` as suggested